src/serde_util.rs

```diff
@@ -67,3 +67,62 @@ where
         (self.parse_fn)(v).map_err(E::custom)
     }
 }
+
+macro_rules! impl_tuple_visitor {
+    ($thing:ident, $len:expr) => {
+        /// Visitor type for de/ser a 32 byte sequence.
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        pub struct $thing<F> {
+            expectation: &'static str,
+            parse_fn: F,
+        }
+
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        impl<F, T, Err> $thing<F>
+        where
+            F: FnOnce(&[u8]) -> Result<T, Err>,
+            Err: fmt::Display,
+        {
+            pub fn new(expectation: &'static str, parse_fn: F) -> Self {
+                $thing {
+                    expectation,
+                    parse_fn,
+                }
+            }
+        }
+
+        #[cfg(any(feature = "std", feature = "alloc"))]
+        impl<'de, F, T, Err> de::Visitor<'de> for $thing<F>
+        where
+            F: FnOnce(&[u8]) -> Result<T, Err>,
+            Err: fmt::Display,
+        {
+            type Value = T;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str(self.expectation)
+            }
+
+            fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
+            where
+                V: de::SeqAccess<'de>,
+            {
+                let mut v = [0u8; $len];
+
+                for i in 0..$len {
+                    if let Some(value) = seq.next_element()? {
+                        v[i] = value;
+                    } else {
+                        let msg = format!("tuple too short, index: {}, expected length: {}", i, $len);
+                        return Err(de::Error::custom(msg));
```

Comment:
`invalid_length()` would be better. Also I think `de::Error` is a bit confusing because it refers to the trait not type. `Err::` would look better but `Err` is also special - even used here - how come it even works? I suggest either calling it `Error` or just `E`.

Reply:
Will do, thanks.